astroquery/esasky/core.py

```diff
@@ -0,0 +1,624 @@
+            os.makedirs(mission_directory)
+        return mission_directory
+    
+    def _extract_file_name_from_response_header(self, headers):
+        content_disposition = headers.get('Content-Disposition')
+        filename_string = "filename="
+        start_index = content_disposition.index(filename_string) + len(filename_string)
+        if (content_disposition[start_index] == "\""):
+            start_index += 1
+       
+        if (self.__FITS_STRING in content_disposition[start_index : ]):
+            end_index = content_disposition.index(self.__FITS_STRING, start_index + 1) + len(self.__FITS_STRING)
+            return content_disposition[start_index : end_index]
+        elif (self.__FTZ_STRING in content_disposition[start_index : ]):
+            end_index = content_disposition.index(self.__FTZ_STRING, start_index + 1) + len(self.__FTZ_STRING)
+            return content_disposition[start_index : end_index]
+        elif (self.__TAR_STRING in content_disposition[start_index : ]):
+            end_index = content_disposition.index(self.__TAR_STRING, start_index + 1) + len(self.__TAR_STRING)
+            return content_disposition[start_index : end_index]
+        else:
+            raise ValueError("Could not find file name in header. Content disposition: %s." %content_disposition)
+            return None
+    
+    def _extract_file_name_from_url(self, product_url):
+        start_index = product_url.rindex("/") + 1
+        return product_url[start_index:]
+    
+    def _query_region_maps(self, coordinates, radius, observation_name, get_query_payload, cache):
+        observation_tap_name = self._find_observation_tap_table_name(observation_name, cache)
+        query = self._build_observation_query(coordinates, radius, self._find_observation_parameters(observation_tap_name, cache))
+        request_payload = self._create_request_payload(query)
+        if(get_query_payload):
+            return request_payload
+        return self._get_and_parse_from_tap(request_payload, cache)
+    
+    def _query_region_catalog(self, coordinates, radius, catalog_name, get_query_payload, cache):
+        catalog_tap_name = self._find_catalog_tap_table_name(catalog_name, cache)
+        query = self._build_catalog_query(coordinates, radius, self._find_catalog_parameters(catalog_tap_name, cache))
+        request_payload = self._create_request_payload(query)
+        if(get_query_payload):
+            return request_payload
+        return self._get_and_parse_from_tap(request_payload, cache)
+        
+    def _build_observation_query(self, coordinates, radius, json):
+        raHours, dec = commons.coord_to_radec(coordinates)
+        ra = raHours * 15.0 # Converts to degrees
+        radiusDeg = commons.radius_to_unit(radius, unit='deg')
+        
+        select_query = "SELECT DISTINCT"
+        
+        metadata = json[self.__METADATA_STRING]
+        metadata_tap_names = ""
+        for i in range(len(metadata)): 
```

Comment:
This is a very minor thing, but if you search for the character "i", it has matches in many irrelevant lines of code, variable names, etc.  If you search for "ii", it only matches the counting index.  Alternatively, you could do `for ind in range(...)` or something more explicit.  I just try to avoid using single-character variable names.  Again, minor - don't worry about it if you don't want to change it.